File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    static inline const std::array<std::string, 26> morse{
        ".-", "-...", "-.-.", "-..",".","..-.","--.","....", 
        "..",".---","-.-",".-..","--","-.","---",".--.","--.-",
        ".-.","...", "-", "..-","...-",".--","-..-","-.--","--..",
    };
    
    int uniqueMorseRepresentations(std::vector<std::string>& words) {
        std::unordered_set<std::string> transformations{};
        for (const auto& word : words) {
            std::string transf{};
            for (const auto c : word) 
                transf += morse[c - 'a'];
            transformations.insert(std::move(transf));
        }
        return static_cast<int>(transformations.size());
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.java
class Solution {
   public int uniqueMorseRepresentations(String[] words) {
        String[] arr = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Set<String> set = new HashSet<>();
        for(String word : words){
            String curr = "";
            for(char w : word.toCharArray())
                curr += arr[w-'a'];
            set.add(curr);
        }
        return set.size();
    }
}